router/core/flushwriter.go

```diff
@@ -57,12 +77,19 @@ func (f *HttpFlushWriter) Flush() (err error) {
 	resp := f.buf.Bytes()
 	f.buf.Reset()
 
-	if f.sse {
-		_, err = f.writer.Write([]byte("event: next\ndata: "))
+	flushBreak := GetWriterPrefix(f.sse, f.multipart)
+	if flushBreak != "" {
+		_, err = f.writer.Write([]byte(flushBreak))
 		if err != nil {
 			return err
 		}
 	}
+
+	if f.multipart && len(resp) > 0 {
+		// Per the Apollo docs, multipart messages are supposed to be json, wrapped in ` "payload"`
+		resp = append([]byte(`{"payload":`), resp...)
```

Comment:
For this type of operations, I'd use our astjson library. This ensure that a valid value is constructed without newlines.